ZEP008-hot-backup.md

```diff
@@ -0,0 +1,619 @@
+Hence, it is important to have the monitor api to keep track of the backup.
+
+In each partition, the backup consists of state until the backup is started. The backup will not contain any new data written after the request is acknowledged.
+
+#### Monitor Backups
+
+A user can monitor the status of the backup by sending monitor request to the coordinator.
+The coordinator responds with a status - `doesNotExist | ongoing | completed | failed`.
+
+- `doesNotExist` : TriggerBackup command with this `backupId` was never send.
+- `ongoing` : Backup is currently being taken. Monitor again for the status.
+- `completed`: Backup is taken successfully. The cluster will be able to successfully restore from this backup.
+- `failed`: Backup has failed. A new backup with the same id cannot be taken. When retrying a new `backupId` must be used.
+
+#### Restore
+
+The user must specify the id of the backup from which Zeebe should restore from. Copying the data and restoring the state from it is done either by Zeebe or an accompanying helper application. Users shouldn't have to do anything manually.
+
+### Internal backup process
+
+Backup of a Zeebe cluster consists of backup of all partitions. A backup of a partition consists of a snapshot and the logStream containing the commands and events after the snapshot position. A partition should be able to restart from this backup the same way it restores its state after a failover or after a normal restart.
+
+For the backup process, we introduce the following concepts:
+-  Checkpoint command : This is a command which triggers the backup process with in a partition. A checkpoint can be triggered in two ways.
+    1. By a new record `checkpoint`.
+    2. When a command send by another partition is received by the partition. These commands are the ones send between partitions by the StreamProcessor. They are the commands related to deployment distribution and message correlation.
+- `checkpointId` : This is the id of the backup triggered by the checkpoint command. `checkpointId` must be included in the checkpoint command. CheckpointId is ordered. A backup must have a checkpointId higher than all previous backups.
+- `checkpointPosition` : This is the position of the command in the `logStream` which triggered the backup.
+- `restore x`: A new command to identify the restored position. This is only used when Zeebe is restored from a backup.
+
+#### Highlevel overview of backup process
+
+This is a high level view of the backup process. The detailed process is described in the Reference-level explanation.
+
+##### Coordinator
+Coordinator could be a gateway. Coordinator receives a request to take backup from the user and sends a response back to the user regarding the status of the backup.
+
+On receiving a `TriggerBackup backupId` request:
+- Sends a command `checkpoint backupId` to all partitions
```

Comment:
:question: How do we deal with back pressure here? Should we white list this? What happens if we do on the load?

Reply:
Either way is fine. If we don't whitelist this command, it is ok because retrying the backup with the same id is already handled and would not create extra work in the system.